src/module.c

```diff
@@ -7760,6 +7997,14 @@ void moduleBlockedClientTimedOut(client *c) {
      * explicit call). See #6798. */
     if (bc->unblocked) return;
 
+    /* If custom auth is in progress, reply to the client with an ERR incase it is timed out. */
+    if (c && (c->flags & CLIENT_CUSTOM_AUTH)) {
+        c->flags &= ~CLIENT_CUSTOM_AUTH;
+        c->flags &= ~CLIENT_CUSTOM_AUTH_RESULT;
+        c->prev_custom_auth_ctx = NULL;
+        addAuthErrReply(c, NULL);
```

Comment:
I don't know I fully understand this choice, why would we automatically reply with an authentication error in a timeout scenario?

Reply:
This callback times out a blocking operation. It does not close the client. So, because the blocking operation is in the context of a command, we will need to reply to the client. This is just handling the case where the other blocking APIs are used in the context of a custom authentication.

However, invoking a timeout callback is not possible from a blocking custom auth operation. So, instead of handling it, we can server assert here .

Reply:
We have two options:

1) Prevent the Module from using other Module Blocking APIs (`RM_BlockClient`, `RM_BlockClientOnKeys`, `RM_BlockClientOnKeyFlags`) by adding a `serverAssert` in these functions to ensure custom module authentication is not in progress.

2) Add `serverAssert` in the `moduleBlockedClientTimedOut` to make sure timeouts do not exist in the case of this module blocked client (since the 3 other blocking APIs should not have been used).

Reply:
In this commit (https://github.com/redis/redis/pull/11659/commits/83d7d3c95fb9d2caaab946b82040f310eebcb23f) - we are now preventing Modules from using other Module Blocking APIs (RM_BlockClient, RM_BlockClientOnKeys, RM_BlockClientOnKeyFlags) when custom auth is ongoing by replying with an `ERR` and setting the `module_blocked_handle` to NULL. Modules can then use RM_UnblockClient to clean up